### PR TITLE
feat: add wrapper to samtools collate

### PR DESCRIPTION
**Pull request by fgvieira**
### QC

* [x] I confirm that:

For all wrappers added by this PR, 

* there is a test case which covers any introduced changes,
* `input:` and `output:` file paths in the resulting rule can be changed arbitrarily,
* either the wrapper can only use a single core, or the example rule contains a `threads: x` statement with `x` being a reasonable default,
* rule names in the test case are in [snake_case](https://en.wikipedia.org/wiki/Snake_case) and somehow tell what the rule is about or match the tools purpose or name (e.g., `map_reads` for a step that maps reads),
* all `environment.yaml` specifications follow [the respective best practices](https://stackoverflow.com/a/64594513/2352071),
* the `environment.yaml` pinning has been updated by running `snakedeploy pin-conda-envs environment.yaml` on a linux machine,
* wherever possible, command line arguments are inferred and set automatically (e.g. based on file extensions in `input:` or `output:`),
* all fields of the example rules in the `Snakefile`s and their entries are explained via comments (`input:`/`output:`/`params:` etc.),
* `stderr` and/or `stdout` are logged correctly (`log:`), depending on the wrapped tool,
* temporary files are either written to a unique hidden folder in the working directory, or (better) stored where the Python function `tempfile.gettempdir()` points to (see [here](https://docs.python.org/3/library/tempfile.html#tempfile.gettempdir); this also means that using any Python `tempfile` default behavior works),
* the `meta.yaml` contains a link to the documentation of the respective tool or command,
* `Snakefile`s pass the linting (`snakemake --lint`),
* `Snakefile`s are formatted with [snakefmt](https://github.com/snakemake/snakefmt),
* Python wrapper scripts are formatted with [black](https://black.readthedocs.io).
* Conda environments use a minimal amount of channels, in recommended ordering. E.g. for bioconda, use (conda-forge, bioconda, nodefaults, as conda-forge should have highest priority and defaults channels are usually not needed because most packages are in conda-forge nowadays).

### CODE REVIEW

**Comment by cmeesters** [comment]
@fgvieira happy to review, and I already triggered a merge with main - yet I have a paper to submit today and will find time tomorrow

File: bio/samtools/collate/environment.yaml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+channels:
+  - conda-forge
+  - bioconda
+  - nodefaults
+dependencies:
+  - samtools =1.20
+  - snakemake-wrapper-utils =0.6.2
```
Comment (cmeesters) [inline bio/samtools/collate/environment.yaml:7]: should we not read >= for such cases?

**Review by cmeesters — APPROVED** [review]
for the test 2 threads and otherwise 8 seems fine - it is my experience, too, that samtools scales to about 8 threads max. regardless of the machine (sometimes only up to 2-4)

File: bio/samtools/collate/environment.yaml (hunk 1 above)
Comment (fgvieira) [reply bio/samtools/collate/environment.yaml:7]: It was my impression that versions are "fixed" and are updated through a GitHub action (so every version bump is tested).

Comment (cmeesters) [reply bio/samtools/collate/environment.yaml:7]: ok, cool - did not know that.

Please excuse me, I need to take a week off, definitively going offline, now. So, most reviews which are pending are at @dlaehnemann, right?

Comment (fgvieira) [reply bio/samtools/collate/environment.yaml:7]: No problem and thanks for the reviews.

**Pull request merged**